extensions/kafka-streams/runtime/src/main/java/io/quarkus/kafka/streams/runtime/KafkaStreamsTopologyManager.java

```diff
@@ -80,17 +84,90 @@ private static Properties getStreamsProperties(Properties properties, String boo
+        if (runtimeConfig.schemaRegistryUrl.isPresent()) {
+            streamsProperties.put(runtimeConfig.schemaRegistryKey, runtimeConfig.schemaRegistryUrl.get());
+        }
+
+        // sasl
+        SaslConfig sc = runtimeConfig.sasl;
+        if (sc != null) {
+            setProperty(sc.jaasConfig, streamsProperties, SaslConfigs.SASL_JAAS_CONFIG);
+
+            setProperty(sc.clientCallbackHandlerClass, streamsProperties, SaslConfigs.SASL_CLIENT_CALLBACK_HANDLER_CLASS);
+
+            setProperty(sc.loginCallbackHandlerClass, streamsProperties, SaslConfigs.SASL_LOGIN_CALLBACK_HANDLER_CLASS);
+            setProperty(sc.loginClass, streamsProperties, SaslConfigs.SASL_LOGIN_CLASS);
+
+            setProperty(sc.kerberosServiceName, streamsProperties, SaslConfigs.SASL_KERBEROS_SERVICE_NAME);
+            setProperty(sc.kerberosKinitCmd, streamsProperties, SaslConfigs.SASL_KERBEROS_KINIT_CMD);
+            setProperty(sc.kerberosTicketRenewWindowFactor, streamsProperties,
+                    SaslConfigs.SASL_KERBEROS_TICKET_RENEW_WINDOW_FACTOR);
+            setProperty(sc.kerberosTicketRenewJitter, streamsProperties, SaslConfigs.SASL_KERBEROS_TICKET_RENEW_JITTER);
+            setProperty(sc.kerberosMinTimeBeforeRelogin, streamsProperties, SaslConfigs.SASL_KERBEROS_MIN_TIME_BEFORE_RELOGIN);
+
+            setProperty(sc.loginRefreshWindowFactor, streamsProperties, SaslConfigs.SASL_LOGIN_REFRESH_WINDOW_FACTOR);
+            setProperty(sc.loginRefreshWindowJitter, streamsProperties, SaslConfigs.SASL_LOGIN_REFRESH_WINDOW_JITTER);
+
+            setProperty(sc.loginRefreshMinPeriod, streamsProperties, SaslConfigs.SASL_LOGIN_REFRESH_MIN_PERIOD_SECONDS,
+                    DurationToSecondsFunction.INSTANCE);
+            setProperty(sc.loginRefreshBuffer, streamsProperties, SaslConfigs.SASL_LOGIN_REFRESH_BUFFER_SECONDS,
+                    DurationToSecondsFunction.INSTANCE);
+        }
+
+        // ssl
+        SslConfig ssl = runtimeConfig.ssl;
+        if (ssl != null) {
+            setProperty(ssl.protocol, streamsProperties, SslConfigs.SSL_PROTOCOL_CONFIG);
+            setProperty(ssl.provider, streamsProperties, SslConfigs.SSL_PROVIDER_CONFIG);
+            setProperty(ssl.cipherSuites, streamsProperties, SslConfigs.SSL_CIPHER_SUITES_CONFIG);
+            setProperty(ssl.enabledProtocols, streamsProperties, SslConfigs.SSL_ENABLED_PROTOCOLS_CONFIG);
+
+            setStoreConfig(ssl.truststore, streamsProperties, "ssl.truststore");
+            setStoreConfig(ssl.keystore, streamsProperties, "ssl.keystore");
+            setStoreConfig(ssl.key, streamsProperties, "ssl.key");
+
+            setProperty(ssl.keymanagerAlgorithm, streamsProperties, SslConfigs.SSL_KEYMANAGER_ALGORITHM_CONFIG);
+            setProperty(ssl.trustmanagerAlgorithm, streamsProperties, SslConfigs.SSL_TRUSTMANAGER_ALGORITHM_CONFIG);
+            setProperty(ssl.endpointIdentificationAlgorithm.map(s -> "\"\"".equals(s) ? "" : s), streamsProperties,
```

Comment:
I don't understand why we need that here?

Reply:
There is no other (or better / smarter) way to pass-in an empty string.
(that's what I've been told by @dmlloyd )

And in this case you need to have an empty string, to disable this config/setting.

Reply:
OK. Can't say I'm excited about it but can't think of a better solution. Or maybe we could use `-1` to disable? It's a bit of dark magic but could be documented.

Just a proposal, WDYT?

Reply:
It's a string ... so setting a number ... hmmm, dunno :-)

Reply:
> There is no other (or better / smarter) way to pass-in an empty string.
> (that's what I've been told by @dmlloyd )

No it's not; I definitely never told you that.  If you want a string which can be empty, use `Optional<String>` with `orElse("")`.  If you want an integer which can be empty, use `OptionalInt` (`int` has no native empty representation though, so `orElse` won't help in this case).

> And in this case you need to have an empty string, to disable this config/setting.

Then it should be `Optional`.